tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/ExpectContinueTestCase.java

```diff
@@ -76,6 +78,33 @@ public void test100Continue() {
                 TestUtils.LARGE_ENTITY.getBytes().length);
     }
 
+    @Test(description = "Test ignoring inbound payload with a 417 response for request with expect:100-continue header")
+    public void test100ContinueNegative() {
+        HttpClient httpClient = new HttpClient("localhost", servicePort);
+
+        DefaultHttpRequest httpRequest = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/continue");
+        DefaultLastHttpContent reqPayload = new DefaultLastHttpContent(
+                Unpooled.wrappedBuffer(TestUtils.LARGE_ENTITY.getBytes()));
+
+        httpRequest.headers().set(HttpHeaderNames.CONTENT_LENGTH, TestUtils.LARGE_ENTITY.getBytes().length);
+        httpRequest.headers().set(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.TEXT_PLAIN);
+
+        List<FullHttpResponse> responses = httpClient.sendExpectContinueRequest(httpRequest, reqPayload);
+
+        Assert.assertFalse(httpClient.waitForChannelClose());
+
+        // 417 Expectation Failed response
+        Assert.assertEquals(responses.get(0).status(), HttpResponseStatus.EXPECTATION_FAILED);
+        int length = Integer.parseInt(responses.get(0).headers().get(HttpHeaderNames.CONTENT_LENGTH));
+        Assert.assertEquals(length, 26);
+        Assert.assertEquals(responses.get(0).content()
+                                    .readCharSequence(length, Charset.defaultCharset()).toString(),
```

Comment:
Shall we make it little bit readable by introducing temporary variable and use that in the assert? WDYT?